### DISCUSSION TITLE
Sharing domain with other application causes 404s

### DISCUSSION THREAD

**Discussion by KendallVsCode** [vercel/next.js#39907] [Help]
Hi all,

I've got a NextJS application living under the same domain as a Sitecore application. When there is a link within the NextJS app that refers to a page in Sitecore, the NextJS app router tries to find it within NextJS, then throws a 404. To circumvent that I've added the attributes (rel="noreferrer" and target="_blank") to the offending links but it feels awfully hacky. Is there a more elegant solution for that where by if the page is not found in NextJS it forwards the request to the Sitecore app?

Thanks!

**Comment by icyJoseph** [comment]
Hi,

you should use rewrites with a fallback, as documented here: https://nextjs.org/docs/api-reference/next.config.js/rewrites#incremental-adoption-of-nextjs, that way when a requests for a path doesn't exist in the file system is linked to, you can serve the content of "the other app".

```js
module.exports = {
  async rewrites() {
    return {
      fallback: [
        {
          source: '/:path*',
          destination: `https://the-domain-where-everything-lives/:path*`,
        },
      ],
    }
  },
}

```

**Comment by KendallVsCode** [comment]
Thank you for your reply, but that doesn’t seem to entirely work. For example if I add:

async rewrites() {
    return {
      fallback: [
        {
          source: "/:path*",
          destination: `https://www.google.com/`,
        },
      ],
    };
  },
 
And enter a url that doesn’t exist in NextJS, it seems to “wrap” the Google site instead of redirect there.

**Reply by icyJoseph** [reply]
Mmm perhaps I misunderstood. Could you investigate using this, https://nextjs.org/docs/api-reference/next.config.js/basepath?

**Reply by KendallVsCode** [reply]
I did look at that as well, but I don't think that's going to do it either. The other app (Sitecore) is on a completely different server. It's been a pain, and something unfortunately we only discovered after having gone into a production enviroment. Your initial suggestion is SO CLOSE, I wish it just redirected to the page instead of doing the rewrite and wrapping the results. And the "redirect" that comes with NextJS funtionality doesn't seem to account for 404s.

**Reply by icyJoseph** [reply]
Ah yeah, like Middleware would still apply... It's a bit unfortunate then :( I'll try to think again tomorrow.
